.github/workflows/run-tests.yml

```diff
@@ -17,8 +29,26 @@ jobs:
         run: |
           mvn package
           ./src/test/resources/run-s3-tests.sh
-      - name: Build Docker Image
-        run: ./publish_img.sh
-        env:
-          DOCKER_USER: ${{ secrets.DOCKER_USER }}
-          DOCKER_PASS: ${{ secrets.DOCKER_PASS }}
\ No newline at end of file
+      - name: Login to DockerHub
+        uses: docker/login-action@v1 
+        if: github.event_name != 'pull_request'
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASS }}
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: andrewgaul/s3proxy
+          tags: |
+            type=sha,event=branch
+            type=semver,pattern={{version}}
```

Comment:
This tags the image with the full version when pushing a git tag.

It also tags it with `latest` when pushing a git tag